collectors/base_collector.go

```diff
@@ -41,10 +43,12 @@ func CreateBaseCollector(collectorOptions CollectorOptions) *BaseCollector {
 		Sleeping:                     false,
 		earlyAck:                     false,
 		PollingPeriod:                collectorOptions.PollingPeriod,
+		PollingLimit:                 collectorOptions.PollingLimit,
```

Comment:
I think you can move `PollingLimit` and `SourceSubscription` to the `PubSubCollector struct` since they are not used by the BaseCollector code but strictly for pubsub code.

Reply:
Updated